minerl/herobraine/hero/handlers/agent/observations/inventory.py

```diff
@@ -93,35 +96,29 @@ def from_hero(self, obs):
             assert stack["quantity"] >= 0
             assert stack["metadata"] in range(16)
             if key is not None:
+                if key not in self.items:
+                    key = self._other
```

Comment:
`key in self.items` is guaranteed by `util.get_unique_matching_item_list_id(self.items, type_name, stack['metadata'])`.

Reply:
Good point - I've adjusted the logic such that we return self._other iff key = None and self._other in self.items